src/index.ts

```diff
@@ -5,13 +5,48 @@ type PineconeClientConfiguration = {
   apiKey: string
 }
 
+Error.prepareStackTrace = () => '';
+
+async function streamToArrayBuffer(stream: ReadableStream<Uint8Array>): Promise<Uint8Array> {
+  let result = new Uint8Array(0);
+  const reader = stream.getReader();
+  while (true) { // eslint-disable-line no-constant-condition
+    const { done, value } = await reader.read();
+    if (done) {
+      break;
+    }
+
+    const newResult = new Uint8Array(result.length + value.length);
+    newResult.set(result);
+    newResult.set(value, result.length);
+    result = newResult;
+  }
+  return result;
+}
+
 async function handler(func: Function, args: any) {
   try {
     return await func(args)
   } catch (error) {
     // @ts-ignore
-    const message = error?.response?.data?.message || null
-    throw `PineconeClient: Error calling ${func.name.replace("bound ", "")}: ${error} ${message ? `(${message})` : ''}`
+    if (error && error.response) {
+      // @ts-ignore
+      const body = error.response?.body as ReadableStream
+      const buffer = body && await streamToArrayBuffer(body);
+      const text = buffer && new TextDecoder().decode(buffer);
+
+      try {
+        // Handle "RAW" call errors
+        const json = text && JSON.parse(text);
+        return Promise.reject(new Error(`${json.message}`))
```

Comment:
Is it possible for body to be null/undefined?  If so, I think its possible that json is undefined here, and this will throw another error 
`json?.message` would help